MapboxMobileEventsTests/MMEAPIClientTests_cedar.mm

```diff
@@ -0,0 +1,327 @@
+                    [[NSUserDefaults standardUserDefaults] synchronize];
+                    
+                    [apiClient postEvent:event completionHandler:nil];
+                    
+                    expectedURLString = [NSString stringWithFormat:@"%@/%@?access_token=%@", MMEAPIClientBaseURL, MMEAPIClientEventsPath, stagingAccessToken];
+                });
+                
+                it(@"should receive processRequest:completionHandler", ^{
+                    sessionWrapperFake should have_received(@selector(processRequest:completionHandler:));
+                });
+                
+                it(@"should be created properly", ^{
+                    sessionWrapperFake.request.URL.absoluteString should equal(expectedURLString);
+                });
+                
+                afterEach(^{
+                    [[NSUserDefaults standardUserDefaults] removeObjectForKey:MMETelemetryStagingAccessToken];
+                    [[NSUserDefaults standardUserDefaults] synchronize];
+                }); 
+            });
+        });
+        
+        context(@"when posting two events", ^{
+            __block MMEEvent *eventTwo;
+            __block NSData *uncompressedData;
+            
+            beforeEach(^{
+                MMECommonEventData *commonEventData = [[MMECommonEventData alloc] init];
+                commonEventData.vendorId = @"vendor-id";
+                commonEventData.model = @"model";
+                commonEventData.iOSVersion = @"1";
+                commonEventData.scale = 42;
+                
+                eventTwo = [MMEEvent locationEventWithAttributes:@{} instanceIdentifer:@"instance-id-1" commonEventData:commonEventData];
+                
+                NSArray *events = @[event, eventTwo];
+                
+                NSMutableArray *eventAttributes = [NSMutableArray arrayWithCapacity:events.count];
+                [events enumerateObjectsUsingBlock:^(MMEEvent * _Nonnull event, NSUInteger idx, BOOL * _Nonnull stop) {
+                    if (event.attributes) {
+                        [eventAttributes addObject:event.attributes];
+                    }
+                }];
+                
+                uncompressedData = [NSJSONSerialization dataWithJSONObject:eventAttributes options:0 error:nil];
+                
+                [apiClient postEvents:@[event, eventTwo] completionHandler:nil];
+            });
+            
+            it(@"should equal gzip for HTTPHeader", ^{
+                sessionWrapperFake.request.allHTTPHeaderFields[MMEAPIClientHeaderFieldContentEncodingKey] should equal(@"gzip");
+            });
+            
+            it(@"should have compressed", ^{
+                NSData *data = (NSData *)sessionWrapperFake.request.HTTPBody;
+                data.length should be_less_than(uncompressedData.length);
+            });
+        });
+        
+        context(@"when posting three events", ^{
```

Comment:
We could probably delete this since we already have coverage for the compressed case.